kasa/discover.py

```diff
@@ -502,16 +503,17 @@ def _get_device_instance(
         return device
 
 
-class DiscoveryResult(BaseModel):
-    """Base model for discovery result."""
+class EncryptionScheme(BaseModel):
+    """Base model for encryption scheme of discovery result."""
```

Comment:
I think it'd be better to only perform typing changes w/o other refactorings in this PR.

Reply:
This was required because pydantic v1 doesn't handle it when it has a sub class and forward typing is enabled.  There was something in [their docs](https://docs.pydantic.dev/1.10/usage/postponed_annotations/) about calling `update_forward_refs` but that didn't work so pulling it out seemed the simplest way.

Reply:
Ok :+1: